src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Commands/Get-AzureAsInstanceLog.cs

```diff
@@ -0,0 +1,182 @@
+        {
+            this.AsAzureHttpClient = AsAzureHttpClient;
+            this.TokenCacheItemProvider = TokenCacheItemProvider;
+        }
+
+        protected override AzureContext DefaultContext
+        {
+            get
+            {
+                // Nothing to do with Azure Resource Managment context
+                return null;
+            }
+        }
+
+        protected override void SaveDataCollectionProfile()
+        {
+            // No data collection for this commandlet 
+        }
+
+        protected override void PromptForDataCollectionProfileIfNotExists()
+        {
+            // No data collection for this commandlet 
+        }
+
+        protected override void BeginProcessing()
+        {
+            base.BeginProcessing();
+
+            if (AsAzureClientSession.Instance.Profile.Environments.Count == 0)
+            {
+                throw new PSInvalidOperationException(string.Format(Resources.NotLoggedInMessage, ""));
+            }
+
+            serverName = Instance;
+            Uri uriResult;
+
+            // if the user specifies the FQN of the server, then extract the servername out of that.
+            // and set the current context
+            if (Uri.TryCreate(Instance, UriKind.Absolute, out uriResult) && uriResult.Scheme == "asazure")
+            {
+                serverName = uriResult.PathAndQuery.Trim('/');
+                if (string.Compare(AsAzureClientSession.Instance.Profile.Context.Environment.Name, uriResult.DnsSafeHost, StringComparison.InvariantCultureIgnoreCase) != 0)
+                {
+                    AsAzureClientSession.Instance.SetCurrentContext(
+                        new AsAzureAccount(),
+                        AsAzureClientSession.Instance.Profile.CreateEnvironment(uriResult.DnsSafeHost));
+                }
+            }
+            else
+            {
+                var currentContext = AsAzureClientSession.Instance.Profile.Context;
+                if (currentContext != null
+                    && AsAzureClientSession.AsAzureRolloutEnvironmentMapping.ContainsKey(currentContext.Environment.Name))
+                {
+                    throw new PSInvalidOperationException(string.Format(Resources.InvalidServerName, serverName));
+                }
+            }
+
+            if (this.AsAzureHttpClient == null)
+            {
+                this.AsAzureHttpClient = new AsAzureHttpClient(() =>
+                {
+                    return new HttpClient();
+                });
+            }
+
+            if (this.TokenCacheItemProvider == null)
+            {
+                this.TokenCacheItemProvider = new TokenCacheItemProvider();
+            }
+        }
+
+        protected override void InitializeQosEvent()
+        {
+            // nothing to do here.
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Instance, Resources.RestartingAnalysisServicesServer))
```

Comment:
@AOrazaev the notification you used for this cmdlet (which should be removed when you get rid of `SupportsShouldProcess` indicates that this cmdlet restarts as AS server. Is that what is intended by this cmdlet?